harness/determined/common/experimental/determined.py

```diff
@@ -112,7 +111,17 @@ def whoami(self) -> user.User:
         return self._from_bindings(resp.user)
 
     def logout(self) -> None:
-        bindings.post_Logout(self._session)
+        auth = self._session._auth
+        # auth should only be None in the special login Session, which must not be used in a
+        # Determined object.
+        assert auth, "Determined.logout() found an unauthorized Session"
```

Comment:
why not have this just return?

Reply:
Because I'm like 99.9% certain that would actually be a bug, and I prefer to surface bugs.